gui/widgets/grid.cpp

```diff
@@ -336,13 +373,27 @@ GridWidget::GridWidget(GuiObject *boss, int x, int y, int w, int h)
 	_scrollPos = 0;
 	_firstVisibleItem = 0;
 	_itemsOnScreen = 0;
+	_itemsPerRow = 0;
+
+	_innerHeight = 0;
+	_innerWidth = 0;
+	_scrollWindowHeight = 0;
+	_scrollWindowWidth = 0;
+	_gridYSpacing = 0;
+	_gridXSpacing = 0;
 
-	_selectedEntry = -1;
+	_selectedEntry = nullptr;
 }
 
 GridWidget::GridWidget(GuiObject *boss, const String &name)
 	: ContainerWidget(boss, name), CommandSender(boss) {
 	_iconDir = ConfMan.get("iconspath");
+
+	_thumbnailHeight = g_gui.xmlEval()->getVar("Globals.GridItemThumbnail.Height");
```

Comment:
This is duplicated across the two constructors, which makes maintenance slightly harder, please outfactor to a function.

Reply:
I looked at the other constructor, and it seemed to be redundant. Additionally GridItemWidget also doesn't require a second constructor since it is always initialised by the GridWidget, so I removed it too, in commit 29d95be.